### PR TITLE
PkgTemplates-ify the package + update to work with new ModelingToolkit interface

### PR DESCRIPTION
**Pull request by nicholaskl97**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

Used PkgTemplates to set up GitHub actions and rearrange files, moved demos to test folder, and updated calls to a now-deprecated ModelingToolkit interface.

### CODE REVIEW

File: .JuliaFormatter.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# See https://domluna.github.io/JuliaFormatter.jl/stable/ for a list of options
+style = "blue"
```
Comment (ChrisRackauckas) [inline .JuliaFormatter.toml]:
```suggestion
style = "sciml"
```

File: Manifest.toml [hunk 2]
```diff
@@ -1,29 +1,13 @@
 # This file is machine-generated - editing it directly is not advised
 
-julia_version = "1.9.0"
```
Comment (ChrisRackauckas) [inline Manifest.toml:3]: remove the manifest

Comment (nicholaskl97) [reply Manifest.toml:3]: Done, and I added it to .gitignore

**Comment by nicholaskl97** [comment]
@ChrisRackauckas, do you know why the Documentation CI fails? I think it's set up just how PkgTemplates defaults, which probably isn't how the SciML documentation works anyway, so maybe it doesn't matter.

**Comment by ChrisRackauckas** [comment]
Maybe @ArnoStrouwen has an idea

**Comment by ArnoStrouwen** [comment]
Why does this Project.toml file:
https://github.com/SciML/NeuralLyapunov.jl/pull/13/files#diff-6e0adb5e01dce01db395f52201532ad8fec022922489f34827886053953ce93f
and this Project.toml file:
https://github.com/SciML/NeuralLyapunov.jl/pull/13/files#diff-72ed386c2a0cd1d23c0968297e70023ed98c22490d146dd89fc91f48369bad4d
have a different UUID for NeuralLyapunov?

**Comment by nicholaskl97** [comment]
> Why does this Project.toml file: https://github.com/SciML/NeuralLyapunov.jl/pull/13/files#diff-6e0adb5e01dce01db395f52201532ad8fec022922489f34827886053953ce93f and this Project.toml file: https://github.com/SciML/NeuralLyapunov.jl/pull/13/files#diff-72ed386c2a0cd1d23c0968297e70023ed98c22490d146dd89fc91f48369bad4d have a different UUID for NeuralLyapunov?

That was it. When I originally ran PkgTemplates, it was in a fork of this repo, since I wasn't sure if it would break anything, but I see doing that is what broke something.

**Pull request merged**